editor/scene_tree_editor.cpp

```diff
@@ -645,6 +645,13 @@ void SceneTreeEditor::_rename_node(ObjectID p_node, const String &p_name) {
 	TreeItem *item = _find(tree->get_root(), n->get_path());
 	ERR_FAIL_COND(!item);
 
+	// Nodes must have a non empty name, so resets it to previous text and show warning
+	if (p_name.empty()) {
+		item->set_text(0, n->get_name());
```

Comment:
I don't think this is needed, it's already the existing text normally. You can also move the whole `if` test at the start of the method to avoid needless node fetching and casting.

Reply:
The TreeItem at this points has already an empty name, if I don't reset its text it would remain empty. The Node name instead is still unmodified that's why I use its name to reset the TreeItem text.

If the TreeItem text wasn't modified I could have put this at the start of the method.